File: examples/js/src/007-mail-support-populate-db.ts
import fs from "fs";
import * as lancedb from "@lancedb/lancedb";
import { OllamaEmbeddings } from "@langchain/ollama";

type Document = {
  id: string;
  content: string;
  vector: number[];
};

const embeddings = new OllamaEmbeddings({
  model: "qwen3-embedding:0.6b",
});

async function addDocument(
  files: string[],
  data: Document[],
  index: number = 0,
): Promise<void> {
  const filePath = `../support_mails/${files[index]}`;
  console.log(`Loading document: ${filePath}`);
  const content = fs.readFileSync(filePath, "utf8");
  const vector = await embeddings.embedQuery(content);
  data.push({ id: files[index], content: content, vector: vector });
  if (index < files.length - 1) {
    await addDocument(files, data, index + 1);
  }
}

async function populateData(): Promise<lancedb.Table> {
  const files = fs.readdirSync("../support_mails");
  const data: Document[] = [];
  await addDocument(files, data);
  console.log(`Loaded ${data.length} documents`);
  const db = await lancedb.connect("lancedb.db");
  return await db.createTable("support_mails", data, {
    mode: "overwrite",
  });
  console.log(`Created table with ${data.length} documents`);
}

populateData();

File: examples/python/src/007_mail_support_populate_db.py
import os
import asyncio
import uuid
import lancedb
from langchain_ollama import OllamaEmbeddings

embeddings = OllamaEmbeddings(model="qwen3-embedding:0.6b")


async def add_documents(files: list[str], data: list[dict], index: int = 0):
    file_path = os.path.join("../support_mails", files[index])
    print(f"Loading document: {file_path}")
    with open(file_path, "r") as f:
        content = f.read()
    vector = await embeddings.aembed_query(content)
    data.append({"id": str(uuid.uuid4()), "content": content, "vector": vector})
    if index < len(files) - 1:
        await add_documents(files, data, index + 1)


async def populate_data():
    files = os.listdir("../support_mails")
    data: list[dict] = []
    await add_documents(files, data)
    print(f"Loaded {len(data)} documents")
    db = await lancedb.connect_async("lancedb.db")
    table = await db.create_table("support_mails", data, mode="overwrite")
    print(f"Created table with {len(data)} documents")


asyncio.run(populate_data())
